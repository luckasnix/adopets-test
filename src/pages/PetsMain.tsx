import React, { useState, useEffect, useCallback } from 'react'
import { RouteComponentProps } from 'react-router-dom'
import Loader from '../ui/Loader'
import styles from './PetsMain.module.css'

const apiKey: string = '4bdafbc5-c2cb-4a5a-8932-3bd929de4f18'

const PetsMain: React.FunctionComponent<RouteComponentProps> = (props) => {
  const [isLoading, setIsLoading] = useState<boolean>(true)
  const [accessKey, setAccessKey] = useState<string>('')
  const fetchAccessKey = useCallback(
    async() => {
      await fetch(
        'https://test.adopets.app/v1/auth/session-request',
        {
          method: 'POST',
          headers: { 'Content-type': 'application/json' },
          body: JSON.stringify( { system_api_key: apiKey })
        }
      )
      .then(
        (res: Response) => res.json()
      )
      .then(
        (res: any) => {
          setAccessKey(res.data.access_key)
        }
      )
    },
    []
   )
  useEffect(() => { fetchAccessKey() }, [])
  useEffect(
    () => {
      if(accessKey.length) {
        setIsLoading(false)
        props.history.push('/register', { accessKey })
      }
    },
    [accessKey, props.history]
  )
  return (
    <main className={styles.page}>
      {isLoading ? <Loader/> : null}
    </main>
  )
}

export default PetsMain