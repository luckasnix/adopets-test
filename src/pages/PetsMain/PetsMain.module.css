.page {
  width: 100%;
  height: auto;
  min-height: calc(100vh - 80px);
  display: grid;
  justify-content: center;
  align-content: center;
  background-color: #ce3f71;
}