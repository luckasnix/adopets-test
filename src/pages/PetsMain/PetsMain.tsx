import React, { useState, useEffect, useCallback } from 'react'
import PetsRegister from './containers/PetsRegister/PetsRegister'
import PetsHeader from '../../ui/PetsHeader/PetsHeader'
import PetsLoader from '../../ui/PetsLoader/PetsLoader'
import styles from './PetsMain.module.css'

const apiKey: string = '4bdafbc5-c2cb-4a5a-8932-3bd929de4f18'

const PetsMain: React.FunctionComponent = () => {
  const [isLoading, setIsLoading] = useState<boolean>(true)
  const [accessKey, setAccessKey] = useState<string>('')
  const fetchAccessKey = useCallback(
    async() => {
      await fetch(
        'https://test.adopets.app/v1/auth/session-request',
        {
          method: 'POST',
          headers: { 'Content-type': 'application/json' },
          body: JSON.stringify( { system_api_key: apiKey })
        }
      )
      .then(
        (res: Response) => res.json()
      )
      .then(
        (res: any) => {
          setAccessKey(res.data.access_key)
        }
      )
    },
    []
   )
  useEffect(() => { fetchAccessKey() }, [])
  useEffect(
    () => {
      if(accessKey.length) {
        setIsLoading(false)
      }
    },
    [accessKey]
  )
  return (
    <>
      <PetsHeader/>
      <div className={styles.page}>
        {isLoading ? <PetsLoader/> : <PetsRegister token={accessKey}/>}
      </div>
    </>
  )
}

export default PetsMain