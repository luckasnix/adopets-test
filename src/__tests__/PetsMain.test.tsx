import React from 'react'
import PetsMain from '../pages/PetsMain/PetsMain'
import renderer from 'react-test-renderer'

it(
  'renders correctly',
  () => {
    const tree = renderer.create(<PetsMain/>).toJSON()
    expect(tree).toMatchSnapshot()
  }
)