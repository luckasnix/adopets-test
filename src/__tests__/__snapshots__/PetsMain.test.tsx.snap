// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="logo"
    >
      <svg>
        adopets-logo.svg
      </svg>
    </div>
  </header>,
  <div
    className="page"
  >
    <div
      className="ldsDualRing"
    />
  </div>,
]
`;
